src/main/scala/mesosphere/marathon/api/v2/AppsResource.scala

```diff
@@ -159,9 +159,10 @@ class AppsResource @Inject() (
     * @param body is the raw, unparsed JSON
     * @param partialUpdate true if the JSON should be parsed as a partial application update (all fields optional)
     *                      or as a wholesale replacement (parsed like an app definition would be)
+    * @param newApp if the app is created via put, we shouldn't treat it as a partial update
     */
-  def canonicalAppUpdateFromJson(appId: PathId, body: Array[Byte], partialUpdate: Boolean): raml.AppUpdate = {
-    if (partialUpdate) {
+  def canonicalAppUpdateFromJson(appId: PathId, body: Array[Byte], partialUpdate: Boolean, newApp: Boolean): raml.AppUpdate = {
+    if (partialUpdate && !newApp) {
```

Comment:
this doesn't make sense to me... 
`newApp` says if app is created via put... it isn't a partial update... why wouldn't we just `partialUpdate=False` ?

What does `partialUpdate=True`  AND `newApp=True` mean?